### PR TITLE
Rework - Content moderator dashboard

### LINKED ISSUE #396: Rework: Content Moderator Refactoring
The functionalities that we had last year are still there, they just need to be refactored to match the new database design. Also the components require some refactoring.

### PR DESCRIPTION
**Pull request by v-stamenova**
# Description

It's a big one, buckle up (I'm sorry in advance).

This branch introduces refactoring on the content moderator side - more specifically rework on the Booth, Company, Sponsorship, Rooms, Presentation CRUD. The functionalities are the same just refactored to use the new database.

**Scheduling and Users are not refactored**

Unit tests are introduced to hopefully reduce checking time. Design is the old one, as it's not a priority for the current PR.

closes #396 

## Type of change
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

## What needs to be tested

Test that you cannot reach the following if you are just participant and that you can reach it only as event organizer.

- [ ] Booth CRUD
- [ ] Company CRUD
- [ ] Presentation CRUD
- [ ] Sponsorship CRUD
- [ ] Room CRUD

# Checklist:

- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

As soon as it it closed new issues will arise - reworking the routes for example

### CODE REVIEW

**Review by TimKardol — COMMENTED** [review]
**Presentation CRUD**
Would it be logical to automatically approve the presentation when it is created by the content mod? (I feel like we talked about this earlier and the conclusion was 'no')

**Company CRUD**
Created company is automatically approved?

**Booth CRUD**
Created booth is automatically approved?

File: resources/views/crew/companies/create.blade.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                                <x-label for="rep_email" value="{{ __('Company Representative') }}"></x-label>
+                                <x-select name="rep_email" class="mt-1 block w-full">
+                                    @foreach(User::forCompanyRep()->get() as $user)
+                                        <option value="{{ $user->email }}">
+                                            {{ $user->name }} | {{ $user->email }}
+                                        </option>
+                                    @endforeach
+                                </x-select>
+                                <x-input-error for="rep_email" class="mt-2"></x-input-error>
+                            </div>
```
[the remark this answers is not in the document]
Comment (v-stamenova) [reply resources/views/crew/companies/create.blade.php:86]: Yep yep forgot about it, will fix it in the evening

File: app/Http/Controllers/Crew/CompanyController.php [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    {
+        if (Auth::user()->cannot('create', Company::class)) {
+            abort(403);
+        }
+
+        // Validate and fetch the user
+        $input = $request->validate([
+            'name' => 'required',
+            'description' => 'required',
+            'website' => 'required|url',
```
[the remark this answers is not in the document]
Comment (v-stamenova) [reply app/Http/Controllers/Crew/CompanyController.php:55]: Will reuse the component I made in the registration/login PR and will think about it

**Review by dwaard — APPROVED** [review]
I did a brief review of the changes
The roles and permissions names might be reviewed to match a specific naming convention, but that can be done in the next PR

**Review by TimKardol — APPROVED** [review]

**Pull request merged**